libraries/rush-lib/src/logic/pnpm/PnpmLockfileConfiguration.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { AlreadyReportedError, Encoding } from '@rushstack/node-core-library';
+import { FileSystem } from '@rushstack/node-core-library';
+import yaml from 'js-yaml';
+import type { Lockfile } from '@pnpm/lockfile-types';
+import type { ITerminal } from '@rushstack/terminal';
+
+import type { RushConfiguration } from '../../api/RushConfiguration';
+import { CustomTipId } from '../../api/CustomTipsConfiguration';
+import type { PnpmLockfilePolicy } from './PnpmOptionsConfiguration';
+
+export class PnpmLockfileConfiguration {
+  /**
+   * Determine whether `pnpm-lock.yaml` complies with the rules specified in `common/config/rush/pnpm-config.schema.json`.
+   * @internal
+   */
+  public static validateLockfile(
```

Comment:
Add your validation logic to the existing lockfile validator function here: https://github.com/microsoft/rushstack/blob/8fe0ea8f861a1cecd8b964bcdabc610f2a5574e3/libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts#L340

Reply:
@dmichon-msft 

It seems that `validate` function will not be called when we run `rush update`


![image](https://github.com/microsoft/rushstack/assets/68707557/e34b8c2d-0861-4f95-9d3b-5cc902cbd8d8)

Reply:
But we want to perform validation when the user runs `rush update`. Cc @octogonz